### PR TITLE
Add cucumber test to verify hybrid mined coinbases

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Description
- Added cucumber test to verify hybrid mined coinbases
- Added cucumber methods to do hybrid mining till specified blockchain height
- Updated the mining node to accept a new `--mine-till-height` command line parameter
- Fixed mining node to acknowledge environment variables for `num_mining_threads` and
  `validate_tip_timeout_sec`; this only worked via the config file
- Added number of mining threads for the SHA3 mining node to use as an option
- Cleaned up the `log4rs_sample_mining_node.yml` config file
- Added custom `mining_node.yml` as an input file for cucumber tests
- Fixed flaky test `tx_broadcast_protocol_submit_mined_then_not_mined_resubmit_success`

## Motivation and Context
Hybrid mining cucumber tests were lacing

## How Has This Been Tested?
Ran cucumber test

## Checklist:

* [X] I'm merging against the `tari-script` branch.
* [X] I have squashed my commits into a single commit.

### CODE REVIEW

File: applications/tari_mining_node/src/errors.rs [hunk 1]
```diff
@@ -30,6 +30,8 @@ pub enum MinerError {
     GrpcConnection(#[from] tonic::transport::Error),
     #[error("Node not ready")]
     NodeNotReady,
+    #[error("Blockchain reached specified height {0}, mining will be stopped")]
+    MineTillHeightReached(u64),
```
Comment (stringhandler) [inline applications/tari_mining_node/src/errors.rs]:
```suggestion
    MineUntilHeightReached(u64),
```

File: applications/tari_mining_node/src/main.rs [hunk 2]
```diff
@@ -59,7 +59,10 @@ async fn main_inner() -> Result<(), ExitCodes> {
     let (bootstrap, global, cfg) = init_configuration(ApplicationType::MiningNode)?;
     let mut config = <MinerConfig as DefaultConfigLoader>::load_from(&cfg).expect("Failed to load config");
     config.mine_on_tip_only = global.mine_on_tip_only;
-    debug!("mine_on_tip_only is {}", config.mine_on_tip_only);
+    config.num_mining_threads = global.num_mining_threads;
+    config.validate_tip_timeout_sec = global.validate_tip_timeout_sec;
+    debug!("{:?}", bootstrap);
```
Comment (stringhandler) [inline applications/tari_mining_node/src/main.rs:64]: Are these intended to stay here, or is this debugging code?

File: applications/tari_mining_node/src/main.rs [hunk 3]
```diff
@@ -145,7 +152,7 @@ async fn mining_cycle(
             .as_ref()
             .ok_or_else(|| err_empty("header"))?
             .height;
-        validate_tip(node_conn, height).await?;
+        validate_tip(node_conn, height, bootstrap.mine_till_height).await?;
```
Comment (stringhandler) [inline applications/tari_mining_node/src/main.rs]:
```suggestion
        validate_tip(node_conn, height, bootstrap.mine_until_height).await?;
```

File: applications/tari_mining_node/src/main.rs (hunk 2 above)
Comment (hansieodendaal) [reply applications/tari_mining_node/src/main.rs:64]: Intended to stay there. Those values were not assigned, and the debug statements are to show the startup-config in the log file.

**Review by stringhandler — APPROVED** [review]
Nice

**Pull request merged**